aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -402,9 +402,11 @@ void TensorIterator::compute_types(const TensorIteratorConfig& config) {
       // Casts to outputs by creating temporaries of the correct dtype (if needed)
       if (config.cast_common_dtype_to_outputs_ && op.is_output && op.current_dtype != common_dtype_) {
         op.original_tensor = op.tensor;
-        op.tensor = at::empty_like(op.tensor,
-                                   op.tensor.options().dtype(common_dtype_),
-                                   LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+        AutoNonVariableTypeMode non_variable_type_mode;
+        op.tensor = at::native::empty_like(
```

Comment:
Do we care this no longer gets tracing?  Imo we don't care:  the performance gain of dodging dispatch altogether for `at::native::empty*` is worth missing tracing.  Tracing will still profile the overarching TensorIterator op.  If tracing says some TensorIterator op has weirdly high cpu overhead, we can switch to a focused flame graph (which we should do even if `empty*` gave tracing info, and in such case one of the first things the flame graph would tell us would be:  get rid of dispatch+tracing :P).

Reply:
We did not trace it for the majority of history, we started tracing it a few months back, and we can stop IMO. I agree that it's not providing a lot of value.